include/deal.II/fe/fe_tools.templates.h

```diff
@@ -421,106 +421,110 @@ namespace FETools
-          for (unsigned int base = 0; base < fes.size(); ++base)
-            for (unsigned int m = 0; m < multiplicities[base];
-                 ++m,
-                              comp_start +=
-                              fes[base]->n_components() * do_tensor_product)
-              for (unsigned int local_index = 0;
-                   local_index < fes[base]->n_dofs_per_quad(quad_number);
-                   ++local_index, ++total_index)
-                {
-                  const unsigned int index_in_base =
-                    local_index + fes[base]->get_first_quad_index(quad_number);
-
-                  Assert(comp_start + fes[base]->n_components() <=
-                           retval[total_index].size(),
-                         ExcInternalError());
-                  for (unsigned int c = 0; c < fes[base]->n_components(); ++c)
-                    {
-                      Assert(c < fes[base]
-                                   ->get_nonzero_components(index_in_base)
-                                   .size(),
-                             ExcInternalError());
-                      retval[total_index][comp_start + c] =
-                        fes[base]->get_nonzero_components(index_in_base)[c];
-                    }
-                }
-        }
+      if (fes.size() > 0)
+        for (unsigned int quad_number = 0;
+             quad_number <
+             (dim == 2 ?
+                1 :
+                (dim == 3 ? fes.front()->reference_cell().n_faces() : 0));
+             ++quad_number)
+          {
+            unsigned int comp_start = 0;
+            for (unsigned int base = 0; base < fes.size(); ++base)
+              for (unsigned int m = 0; m < multiplicities[base];
+                   ++m,
+                                comp_start +=
+                                fes[base]->n_components() * do_tensor_product)
+                for (unsigned int local_index = 0;
+                     local_index < fes[base]->n_dofs_per_quad(quad_number);
+                     ++local_index, ++total_index)
+                  {
+                    const unsigned int index_in_base =
+                      local_index +
+                      fes[base]->get_first_quad_index(quad_number);
+
+                    Assert(comp_start + fes[base]->n_components() <=
+                             retval[total_index].size(),
+                           ExcInternalError());
+                    for (unsigned int c = 0; c < fes[base]->n_components(); ++c)
+                      {
+                        Assert(c < fes[base]
+                                     ->get_nonzero_components(index_in_base)
+                                     .size(),
+                               ExcInternalError());
+                        retval[total_index][comp_start + c] =
+                          fes[base]->get_nonzero_components(index_in_base)[c];
+                      }
+                  }
+          }
 
       // 4. Hexes
       if (dim == 3)
```

Comment:
Why don't you have to check for `if (fes.size() > 0)` here as well?

Reply:
Here it is not needed, since `fes.front()` is not accessed.

Reply:
I see. `fes.size()` limits the loop, so the whole hexes-part will be skipped.

You could also put parts 1-4 into one large conditional statement, which will also reflect that we really skip everything at first glance. But that's just a suggestion - I don't insist on this :-)

Reply:
Or, since the default constructor is deleted now, place an assertion that `fes.size() > 0` and remove the conditional.

Reply:
The function takes a vector of FEs; so the default constructor of FESystem is somewhat unrelated (and just pointed out the missing check in this function). I am not sure if the function should work for an empty vector. But in the doubt, I would like to keep it in such a way that it can work under such condition.